contrib/check-for-api-changes.sh

```diff
@@ -81,6 +82,13 @@ need_nightly() {
     fi
 }
 
+need_cargo_public_api() {
+    if command -v cargo-public-api > /dev/null; then
+        return
+    fi
+    err "cargo-public-api is not installed; please run 'cargo install cargo-public-api' or 'cargo binstall cargo-public-api'"
```

Comment:
Note: do we add `+nightly` as @yancyribbens reports [here](https://github.com/rust-bitcoin/rust-bitcoin/issues/3764#issuecomment-2542451548) so this becomes:

```suggestion
    err "cargo-public-api is not installed; please run 'cargo +nightly install --locked cargo-public-api'"
```

Yeah I think this is fair. Will do the changes...

Reply:
Sorry I don't think `+nightly` was actually required.  Not sure it really matters either way.

Reply:
Oh ok!